s3api/utils/utils.go

```diff
@@ -83,6 +84,22 @@ func SetResponseHeaders(ctx *fiber.Ctx, headers []CustomHeader) {
 	}
 }
 
+func IsValidBucketName(bucket string) bool {
+	if len(bucket) < 3 || len(bucket) > 63 {
+		return false
+	}
+	// Checks to contain only digits, lowercase letters, dot, hyphen.
+	// Checks to start and end with only digits and lowercase letters.
+	if !regexp.MustCompile(`^[a-z0-9][a-z0-9.-]+[a-z0-9]$`).MatchString(bucket) {
+		return false
+	}
+	// Checks not to be a valid IP address
+	if regexp.MustCompile(`^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$`).MatchString(bucket) {
+		return false
+	}
```

Comment:
put regexp.MustCompile(...) into a global variable, and then use var.MatchString(bucket) in here.  this will speed up the check to only compile the regexp once.